plugins/eventfilter_parser/eventfilter_parser_grammar.leg

```diff
@@ -0,0 +1,199 @@
+%{
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ *    Copyright 2023 (c) Fraunhofer IOSB (Author: Florian Düwel)
+ */
+
+#include "open62541/plugin/eventfilter_parser_grammar.h"
+
+#define YY_CTX_LOCAL 1
+#define YY_CTX_MEMBERS                  \
+    UA_Parsed_EventFilter parsedFilter; \
+    UA_String input;                    \
+    UA_UInt32 pos;
+
+#define YY_INPUT(yyctx, buf, result, max_size) \
+{                                              \
+  if(yyctx->input.length <= yyctx->pos) {      \
+      result = 0;                              \
+  } else {                                     \
+      result = yyctx->input.length - yyctx->pos;\
+      if(result > max_size)                     \
+          result = max_size;                   \
+      memcpy(buf, &yyctx->input.data[yyctx->pos], result); \
+      yyctx->pos += result;                    \
+  }                                            \
+}
+
+UA_Counters ctr;
```

Comment:
Global variables are dangerous.
It could be that a user of the open62541 uses parallel threads.
Then we would have to use locks within the parsing to protect against parallel access of the parsing.
It would be much more elegant, if these global variables were part of some individual "context" that is passed into the internal parsing.